### ISSUE TITLE
find: unknown predicate `-not -path '

### ISSUE THREAD

**Issue by amanagr** [codecov/codecov-python#250]
We see this error on CircleCI builds: eg: https://app.circleci.com/pipelines/github/zulip/zulip/7488/workflows/7906c5d1-5032-47a1-88f3-60e79a9631d1/jobs/43293/parallel-runs/0/steps/0-114

```
find: unknown predicate `-not -path './bower_components/**' -not -path './node_modules/**' -not -path './vendor/**''
 
    Error running `['find', '/home/circleci/zulip', "-not -path './bower_components/**' -not -path './node_modules/**' -not -path './vendor/**'", '-type', 'f', '-name', '*.gcno', '', '-exec', 'gcov', '-pb', '', '{}', '+']`: Command '['find', '/home/circleci/zulip', "-not -path './bower_components/**' -not -path './node_modules/**' -not -path './vendor/**'", '-type', 'f', '-name', '*.gcno', '', '-exec', 'gcov', '-pb', '', '{}', '+']' returned non-zero exit status 1.
```

When running:
```
pip install codecov && codecov
```

Pinning the version of `codecov==2.0.15` fixes this (since it doesn't use `find`.)

We use `buildpack-deps:bionic-scm` as our base image for CI builds.

Our CI config file - https://github.com/zulip/zulip/blob/master/.circleci/config.yml

Let me know if you need any more details to debug this.

**Comment by timabbott** [comment]
To provide some added detail: I think that the `find` syntax used here just unconditionally fails in Ubuntu Bionic.

**Comment by zhuyifei1999** [comment]
This is from https://github.com/codecov/codecov-python/commit/9f50cca7e5c5e0bb5e2859db238da08e3195e105#diff-ed4cb86e1f4a5c5feeecc37b90ec6a23R679

`dont_search_here` has multiple arguments and is expected to be shell-expanded. CC @drazisil

**Comment by drazisil** [comment]
@zhuyifei1999 Changes where made because using the shell opens the app to the possibility of command injection. If you are aware of a secure way to do this, PRs are welcome.

**Comment by nmoinvaz** [comment]
#217 removes find command entire using python to do file searching and excluding.

**Comment by drazisil** [comment]
@nmoinvaz Thanks, taking a look

**Comment by drazisil** [comment]
Should be resolved by #217

**Comment by amanagr** [comment]
Thanks for fixing this quickly!!
